src/api/MarkerBasedAPI.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * @flow
+ * @file class for Box marker based API's to inherit common functionality from
+ * @author Box
+ */
+import Base from './Base';
+
+type Params = {
+    marker: string,
+    limit: number
+};
+
+type Data = {
+    next_marker: string,
+    entries: Array<any>
+};
+
+class MarkerBasedApi extends Base {
+    /**
+     * @property {Data}
```

Comment:
do we need to double up on docs like this (type 2 both places)? Kinda verbose if we're not going to add a comment with any context as to what the property is for

Reply:
The type definition is there above 
```
type Data = {
    next_marker: string,
    limit: number,
    entries: Array<any>
};
```

Reply:
Similarly it's there for OffsetBasedAPI ```
type Data = {
    entries: Array<any>,
    total_count: number
};
```